packages/fx-core/src/common/local/localEnvManager.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import {
+  ConfigFolderName,
+  InputConfigsFolderName,
+  Json,
+  ProjectSettings,
+  ProjectSettingsFileName,
+  UserError,
+} from "@microsoft/teamsfx-api";
+import * as fs from "fs-extra";
+import * as path from "path";
+
+import { LocalSettingsProvider } from "../localSettingsProvider";
+import { waitSeconds } from "../tools";
+import { LocalCrypto } from "../../core/crypto";
+import { CoreSource, ReadFileError } from "../../core/error";
+
+class LocalEnvManager {
+  public async getLaunchInput() {}
+
+  public async getLocalDebugEnvs() {}
+
+  public async getNpmInstallLogInfo() {}
+
+  public async getPortsInUse() {}
+
+  public async getProgrammingLanguage() {}
+
+  public async getSkipNgrokConfig() {}
+
+  private async getLocalSettings(projectPath: string): Promise<Json | undefined> {
+    const projectSettings = await this.getProjectSettings(projectPath);
+    const localSettingsProvider = new LocalSettingsProvider(projectPath);
+    const crypto = new LocalCrypto(projectSettings.projectId);
+    return await this.retry(async () => {
+      return await localSettingsProvider.loadV2(crypto);
+    });
+  }
+
+  private async getProjectSettings(projectPath: string): Promise<ProjectSettings> {
```

Comment:
Do we need to use loadProjectSettings in packages\fx-core\src\core\middleware\projectSettingsLoader.ts?
There are some compatible logic there.

Reply:
No, I'm trying to make `LocalEnvMananger` as independent as possible from `core`. `projectSettingsLoader` is tightly bound on core's lifecycle and context so hard to be used directly.
BTW, `LocalEnvManager` is read-only on project-level settings, so simple `readJson` is enough, no need to consider compatible logic.

Reply:
Got it, thx!